### PR TITLE
julia-syntax.scm: Remove deprecation warning

### LINKED ISSUE #19891: 0 .< A .< 1 gives deprecation warning in 0.6
on 0.6.0-dev.1950 (Win 64), 
`0 .< randn(2,3) .< 1`
gives a warning about A & B being deprecated, and suggests using A .& B instead

### PR DESCRIPTION
**Pull request by Shade5**
Changed & to .& to remove 0 .< A .< 1
deprecation warning

Fixes #19891

### CODE REVIEW

**Comment by Shade5** [comment]
@nanosoldier `runbenchmarks(ALL)`

**Comment by Shade5** [comment]
Won't the old test satisfy this too?

**Comment by TotalVerb** [comment]
@Shade5 I think the reason this has not been caught before was because there was _no_ old test. It is a good time to add one.

**Comment by tkelman** [comment]
you don't have to test for the warning, just write a correctness test that chained comparison with at least one array input gives the output it should

> and write a test case that would have triggered the warning before on master but doesn't after your change

**Comment by Shade5** [comment]
In which file of `test/` should I add the test? How do I find out where to add to if I have to add another test in the future?

**Comment by tkelman** [comment]
I would add it to test/operators.jl.

File: test/operators.jl [hunk 1]
```diff
@@ -89,3 +89,9 @@ end
     @test filter(!isupper, str) == replace(str, r"[A-Z]", "")
     @test filter(!islower, str) == replace(str, r"[a-z]", "")
 end
+
+# issue #19891
+B=0 .< [1 -1 5] .< 3
```
Comment (tkelman) [inline test/operators.jl]: would be a little easier to read with spaces around the =

File: test/operators.jl [hunk 2]
```diff
@@ -89,3 +89,9 @@ end
...
+@test ifelse(B, 1, 2) == [1 2 2]
```
Comment (tkelman) [inline test/operators.jl]: didn't realize ifelse is vectorized like this, probably just test the value and type of B

**Comment by ararslan** [comment]
Just as an aside, only JuliaLang members can run Nanosoldier.

**Comment by Shade5** [comment]
@ararslan Thanks, I didn't know that. The docs of nanosoldier didn't mention that too

**Comment by KristofferC** [comment]
It does. 

"If you're a collaborator in the JuliaLang/julia repository, you can submit CI jobs to the Julia Lab's Nanosoldier cluster at MIT by commenting on commits or pull requests."

I guess the term" collaborator" is a bit fuzzy.

**Comment by Shade5** [comment]
Yup, I thought collaborator meant anyone who was contributing. Got it now, thanks :+1:

File: test/operators.jl [hunk 3]
```diff
@@ -89,3 +89,9 @@ end
     @test filter(!isupper, str) == replace(str, r"[A-Z]", "")
     @test filter(!islower, str) == replace(str, r"[a-z]", "")
 end
+
+# issue #19891
+B = 0 .< [1 -1 5] .< 3
+@test B == [true false false]
+B = 3 .> [1 -1 5] .> 0
+@test B == [true false false]
```
Comment (TotalVerb) [inline test/operators.jl]: I guess Travis doesn't check for this, but it's usually a good idea to have a newline character at the end of a text file.

File: test/operators.jl [hunk 4]
```diff
@@ -89,3 +89,9 @@ end
     @test filter(!isupper, str) == replace(str, r"[A-Z]", "")
     @test filter(!islower, str) == replace(str, r"[a-z]", "")
 end
+
+# issue #19891
+B = 0 .< [1 -1 5] .< 3
```
Comment (TotalVerb) [inline test/operators.jl]: Best to put these tests in a `@testset`; see the test above for an example.

**Comment by stevengj** [comment]
@nanosoldier `runbenchmarks(ALL, vs = ":master")`

**Comment by nanosoldier** [comment]
[Your benchmark job](https://github.com/JuliaLang/julia/pull/19928#issuecomment-271364325) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/BaseBenchmarkReports/blob/15c01a56829b2711f5b89fe893be13600428f8ec/906a7d1_vs_7c34d69/report.md). cc @jrevels

**Comment by Shade5** [comment]
The performance is reducing as `.<` is being used even when `<` should be used right? Should I add a case statement to check for that first?

**Comment by TotalVerb** [comment]
@Shade5 My guess is that all of those regressions are noise.

**Comment by Shade5** [comment]
So I don't have to do anything?

**Comment by stevengj** [comment]
For scalars, `.&` and `.<` should be equivalent to `&` and `<`, so that shouldn't hurt performance.

However, it's an awful lot of regressions for them all to be noise.  Let's run it again and see if they are reproduced:

@nanosoldier `runbenchmarks(ALL, vs = ":master")`

**Comment by TotalVerb** [comment]
While it's not strictly related to this PR, I wonder if something can be done to reduce noise... for example running known noisy benchmarks for a longer period of time.

**Comment by TotalVerb** [comment]
Note also that this PR only applies to chained .< operations, which are not even tested and should be completely unrelated to the regressing benchmarks. A good idea to double check, in any event.

**Comment by nanosoldier** [comment]
[Your benchmark job](https://github.com/JuliaLang/julia/pull/19928#issuecomment-271636404) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/BaseBenchmarkReports/blob/679738e40d315ac7bd0d22bee64cd9c6607d5d97/1d9864b_vs_0316883/report.md). cc @jrevels

**Comment by ararslan** [comment]
Those benchmark results look fine to me; there's only a single regression and it didn't come up in the previous run, so I'm inclined to think it's noise.

**Comment by Shade5** [comment]
Anything else?

**Comment by tkelman** [comment]
Thanks for the contribution!

**Pull request merged**